### PR TITLE
Update Streaming Avro Record Output

### PR DESCRIPTION
**Pull request by kevinjnguyen**
PR with the progress made towards debugging the streaming implementation.

Changes:
* Reverted the revert for Publish Time. The implementation was correct, just missing one more step.
* Added the missing step for publish time. When reading the stream, a flag is passed to consider adding the publish time. 
  * In the execution phase, no publish time (false).
  * In the preparation phase, publish time (true).

### CODE REVIEW

File: crates/sparrow-arrow/src/avro/avro_arrow.rs [hunk 1]
```diff
@@ -13,8 +13,13 @@ use std::sync::Arc;
 /// represents a row and contains tuples with a string field name and the corresponding Avro Value,
 /// and returns a Result containing a vector of Arrow arrays (Vec<ArrayRef>) if the
 /// conversion is successful, or an ArrowError if an error occurs during conversion.
-pub fn avro_to_arrow(values: Vec<Vec<(String, Value)>>) -> Result<Vec<ArrayRef>, ArrowError> {
-    (0..values[0].len())
+pub fn avro_to_arrow(values: Vec<Vec<(String, Value)>>, should_include_publish_time: bool) -> Result<Vec<ArrayRef>, ArrowError> {
+    let upper_limit = if should_include_publish_time {
```
Comment (bjchambers) [inline crates/sparrow-arrow/src/avro/avro_arrow.rs:17]: For long term -- this logic seems error-prone. Rather than relying on offsets (and order of columns) and boolean flags, it seems like it would be better to use the schema and the presence/absence of a given field.

**Review by epinzur — APPROVED** [review]

File: crates/sparrow-runtime/src/execute/compute_executor.rs [hunk 2]
```diff
@@ -245,7 +243,6 @@ impl ComputeExecutor {
 
                 Ok(ProgressUpdate::ExecutionComplete { compute_snapshots })
             };
-
```
Comment (bjchambers) [inline crates/sparrow-runtime/src/execute/compute_executor.rs:248]: Looks like a lot of unrelated changes -- removing empty lines? Probably don't want to include those changes.

File: crates/sparrow-runtime/src/metadata/raw_metadata.rs [hunk 3]
```diff
@@ -171,6 +181,7 @@ impl RawMetadata {
     /// Create a `RawMetadata` from a Pulsar topic.
     pub(crate) async fn try_from_pulsar(
         config: &PulsarConfig,
+        should_include_publish_time: bool,
```
Comment (bjchambers) [inline crates/sparrow-runtime/src/metadata/raw_metadata.rs:184]: I think this can just be `include_publish_time` since it controls the behavior. I don't think `should_` adds much.

**Review by bjchambers — APPROVED** [review]
Seems fine for getting things moving.

Ideally, we'd either introduce an enum for the boolean or we'd simplify things so we didn't need to pass it (possibly deciding based on the presence of the `_publish_time` field in the schema that we need to add it). If we did the enum, it would let us write something `PublishTime::Add` or `PublishTime::None` rather than `true` and `false`, which would make it clearer what the booleans mean, especially in the cases of the functions that take multiple booleans.
